doc/design/new-user-questionnaire.md

```diff
@@ -0,0 +1,56 @@
+# Premise
+Currently, anyone with a GitHub account can signup for DANDI and make full use of its functionality. The goal of this design document is to introduce an
+approval process to help prevent misuse of the archive.
+
+## 1. Implementation Details
+### Changes to User model
+Three new columns, `status`, `questionnaire_form`, and `rejection_reason` will be introduced to the `User` model in dandi-api.
+- `status` is an enum that can be one of four values: `INCOMPLETE`, `PENDING`, `APPROVED`, or `REJECTED`.
+- `questionnaire_form` is a JSON field. The keys are questions, and the values are the users' answers.
+- `rejection_reason` is an optional string that contains a message written by one of the admins describing why a user was denied approval (if applicable).
+
+### Changes to API
+- /users/me needs to include the user's current status
+- A [Django group](https://docs.djangoproject.com/en/3.2/topics/auth/default/#groups), `approved_users`, will be created. Only users in this group will have access to protected endpoints.
+- Only approved users can create and edit dandisets.
+
+### Changes to frontend
+- Configuration so the frontend knows where to find SSR pages
+- User avatar circley thing needs to show user's current status
+
+### New views
+- SSR rendered pages for questionnaire form, pending approval page, rejected page, and admin approval page.
+
+## 2. High-level Overview
+### Current user signup flow
+1) The user navigates to https://gui.dandiarchive.org and clicks the "LOG IN WITH GITHUB" button
+2) The user is redirected to GitHub, enters their credentials, and clicks "Sign in". This action creates a new DANDI Archive account for the user and gives
+them full capabilities of any other normal user (creating/editing dandisets, etc).
+3) The user is redirected back to https://gui.dandiarchive.org signed in to their new account.
+
+### Proposed user signup flow
+1) The user navigates to https://gui.dandiarchive.org and clicks the "LOG IN WITH GITHUB" button.
+2) The user is redirected to GitHub, enters their credentials, and clicks "Sign in". This action creates a new DANDI archive account for the user with the `status` field set to `INCOMPLETE`.
+3) The user is redirected to the SSR rendered questionnaire form and they fill it out and submit it. This populates the `questionnaire_form` field in the DB and sets the `status` field to `PENDING`.
+4) A message is displayed telling the user when they can expect approval.
+5) The user is redirected to the DANDI Archive homepage. The "LOG IN WITH GITHUB" button is now red and says something like "PENDING ACCOUNT APPROVAL". If the
```

Comment:
I guess it relates to "User avatar circley thing needs to show user's current status" [above](https://github.com/dandi/dandi-api/pull/496/files#diff-48ab438eb0621a2dc7ee9fa4f116a6b492f0dbb555fde73b90bfb4a438fb1109R19) making it into a long squarish circley ;)

But that squarish circley should be like an active user circley then since we should still allow users to log out (they are already logged in via github).  So I wonder if it should just remain a circley with initials, but be just of different  color (not blue) to reflect status + that modal window which now shows API key instead providing details on the status (not the key) if clicked upon.  That would make UI more consistent IMHO, and also give place later possibly to add some "Contact us if waited for so long" button within that modal etc.

Reply:
I think we'd use something like the red status dot that Github and other services use. Color would be fine too. This is a question for @jtomeck to help us out with.

Reply:
> I think we'd use something like the red status dot that Github and other services use. Color would be fine too. This is a question for @jtomeck to help us out with.

I agree with you, @waxlamp. The status dot is a nice, simple, yet highly effective UX choice.

Reply:
So I think the regular "logged in" circle with initials would stay, plus the red notification dot (with a popover or something explaining why it's red). And if an unverified user is logged in and tries to do something that requires verification (edit metadata, upload a new dandiset, etc.) then the appropriate button would be grayed out, with a popover message saying they need to be verified, and to either fill out the questionnaire or email the admins (depending on what steps they've completed so far).

Reply:
above sounds good, I guess should be put into text

Reply:
:+1:  Added to design doc https://github.com/dandi/dandi-api/commit/bebba7adadd193820bf24acd6eb368ed27cce68e

Reply:
sounds good - thank you @mvandenburgh